src/event.rs

```diff
@@ -641,6 +649,36 @@ where
 							if allow_0conf { " trusted" } else { "" },
 							counterparty_node_id,
 						);
+						let peer_node_ids = self.peer_manager.get_peer_node_ids();
+						if let Some(peer) = peer_node_ids
+							.iter()
+							.find(|(node_id, _)| node_id == &counterparty_node_id)
+						{
+							let address = match peer.1.clone() {
```

Comment:
yes was just wondering with the code why its trying to reconnect to the wrong port. 
trying to access `self.network_graph.read_only().nodes` it return an empty map, do I need to setup gossip somewhere?

Reply:
been trying to resolve this, will get back to this tomorrow. 
a couple of questions please:

1) I feel there is an error in `rust-lightnin`
in the following logs, the port showing in `node_b_peers` is not in the `node_a` listening addresses array.
in this test im not running my code, this is just starting 2 nodes and creating a channel and then listing peers without stopping any node. 
```---- test::functional_tests::persist_peer_from_inbound_channel stdout ----
== Node A ==
Setting network: regtest
Setting random LDK storage dir: /tmp/xrPtejR
Setting random LDK listening addresses: [TcpIpV4 { addr: [127, 0, 0, 1], port: 9937 }, TcpIpV4 { addr: [127, 0, 0, 1], port: 57309 }]

== Node B ==
Setting network: regtest
Setting random LDK storage dir: /tmp/uu1Os5I
Setting random LDK listening addresses: [TcpIpV4 { addr: [127, 0, 0, 1], port: 12888 }, TcpIpV4 { addr: [127, 0, 0, 1], port: 14086 }]
Generating 101 blocks... Done!

Generating 1 blocks... Done!


A -- connect_open_channel -> B
029fca8b65b2b981ca69872a8ec09efd5d1b618e179a63d05a1dbc460b91d6d945 got event ChannelPending { channel_id: ChannelId([70, 215, 99, 34, 66, 47, 198, 223, 183, 31, 221, 47, 60, 193, 73, 26, 144, 183, 150, 22, 160, 33, 188, 229, 222, 222, 250, 236, 216, 40, 60, 92]), user_channel_id: UserChannelId(189294706845638377905056823224737092261), former_temporary_channel_id: ChannelId([180, 20, 248, 6, 111, 228, 250, 185, 45, 159, 200, 208, 14, 64, 2, 62, 15, 102, 60, 100, 168, 118, 239, 64, 165, 230, 199, 102, 126, 70, 42, 192]), counterparty_node_id: PublicKey(25fa24b0a53b330dd5e3ebc3a822c1b2cc4b9f5be27667a7b30bf37ad6061bec0d4b751e8f47b46ac7c1b07e5ca045347febf1a78328b974d69a57d6093ecd95), funding_txo: OutPoint { txid: 5d3c28d8ecfadedee5bc21a01696b7901a49c13c2fdd1fb7dfc62f422263d746, vout: 1 } }
node_b got event ChannelPending { channel_id: ChannelId([70, 215, 99, 34, 66, 47, 198, 223, 183, 31, 221, 47, 60, 193, 73, 26, 144, 183, 150, 22, 160, 33, 188, 229, 222, 222, 250, 236, 216, 40, 60, 92]), user_channel_id: UserChannelId(244287129025742860562500776048931337989), former_temporary_channel_id: ChannelId([180, 20, 248, 6, 111, 228, 250, 185, 45, 159, 200, 208, 14, 64, 2, 62, 15, 102, 60, 100, 168, 118, 239, 64, 165, 230, 199, 102, 126, 70, 42, 192]), counterparty_node_id: PublicKey(45d9d6910b46bc1d5ad0639a178e611b5dfd9ec08e2a8769ca81b9b2658bca9fe6f20a47262070c9128a9623d4e7cf46d7fdb21da90d1a54a1c75003ffad9194), funding_txo: OutPoint { txid: 5d3c28d8ecfadedee5bc21a01696b7901a49c13c2fdd1fb7dfc62f422263d746, vout: 1 } }

 .. generating blocks ..
Generating 6 blocks... Done!

029fca8b65b2b981ca69872a8ec09efd5d1b618e179a63d05a1dbc460b91d6d945 got event ChannelReady { channel_id: ChannelId([70, 215, 99, 34, 66, 47, 198, 223, 183, 31, 221, 47, 60, 193, 73, 26, 144, 183, 150, 22, 160, 33, 188, 229, 222, 222, 250, 236, 216, 40, 60, 92]), user_channel_id: UserChannelId(189294706845638377905056823224737092261), counterparty_node_id: Some(PublicKey(25fa24b0a53b330dd5e3ebc3a822c1b2cc4b9f5be27667a7b30bf37ad6061bec0d4b751e8f47b46ac7c1b07e5ca045347febf1a78328b974d69a57d6093ecd95)) }
node_b got event ChannelReady { channel_id: ChannelId([70, 215, 99, 34, 66, 47, 198, 223, 183, 31, 221, 47, 60, 193, 73, 26, 144, 183, 150, 22, 160, 33, 188, 229, 222, 222, 250, 236, 216, 40, 60, 92]), user_channel_id: UserChannelId(244287129025742860562500776048931337989), counterparty_node_id: Some(PublicKey(45d9d6910b46bc1d5ad0639a178e611b5dfd9ec08e2a8769ca81b9b2658bca9fe6f20a47262070c9128a9623d4e7cf46d7fdb21da90d1a54a1c75003ffad9194)) }
[src/test/functional_tests.rs:57] &node_b_peers = PeerDetails {
    node_id: PublicKey(
        45d9d6910b46bc1d5ad0639a178e611b5dfd9ec08e2a8769ca81b9b2658bca9fe6f20a47262070c9128a9623d4e7cf46d7fdb21da90d1a54a1c75003ffad9194,
    ),
    address: TcpIpV4 {
        addr: [
            127,
            0,
            0,
            1,
        ],
        port: 43256,
    },
    is_persisted: false,
    is_connected: true,
}
```
2)  when the event `OpenChannelRequest` fired, the network_graph is not updated yet, so iam actually not sure we can rely on that.. 
when moving the code the `ChannelReady`, I do see the node info and announcements, but this is still a bit tricky because we only listen to announcements from peers who we have channels with.

Maybe we should tackle this here https://github.com/lightningdevkit/ldk-node/blob/main/src/lib.rs#L552 ?